File: questoes_entrevistas/validacao_informacoes.py
def validar_nome(nome):
    return len(nome) > 3

def validar_idade(idade):
    return 0 <= idade <= 150

def validar_salario(salario):
    return salario > 0

def validar_sexo(sexo):
    return sexo.upper() in ['M', 'F']

def validar_estado_civil(estado):
    return estado.upper() in ['S', 'C', 'V', 'D']

def main():
    # Validação do nome
    while True:
        nome = input("Digite o nome (mais que 3 caracteres): ")
        if validar_nome(nome):
            break
        print("Nome inválido! Deve ter mais que 3 caracteres.")

    # Validação da idade
    while True:
        try:
            idade = int(input("Digite a idade (entre 0 e 150): "))
            if validar_idade(idade):
                break
            print("Idade inválida! Deve estar entre 0 e 150.")
        except ValueError:
            print("Por favor, digite um número válido.")

    # Validação do salário
    while True:
        try:
            salario = float(input("Digite o salário (maior que zero): "))
            if validar_salario(salario):
                break
            print("Salário inválido! Deve ser maior que zero.")
        except ValueError:
            print("Por favor, digite um número válido.")

    # Validação do sexo
    while True:
        sexo = input("Digite o sexo (M/F): ")
        if validar_sexo(sexo):
            break
        print("Sexo inválido! Digite M ou F.")

    # Validação do estado civil
    while True:
        estado_civil = input("Digite o estado civil (S/C/V/D): ")
        if validar_estado_civil(estado_civil):
            break
        print("Estado civil inválido! Digite S, C, V ou D.")

    print("\nInformações validadas com sucesso!")
    print(f"Nome: {nome}")
    print(f"Idade: {idade}")
    print(f"Salário: {salario}")
    print(f"Sexo: {sexo.upper()}")
    print(f"Estado Civil: {estado_civil.upper()}")

if __name__ == "__main__":
    main()

File: questoes_entrevistas/ValidacaoInformacoes.java
import java.util.Scanner;

public class ValidacaoInformacoes {
    private static boolean validarNome(String nome) {
        return nome.length() > 3;
    }

    private static boolean validarIdade(int idade) {
        return idade >= 0 && idade <= 150;
    }

    private static boolean validarSalario(double salario) {
        return salario > 0;
    }

    private static boolean validarSexo(char sexo) {
        return sexo == 'M' || sexo == 'F' || sexo == 'm' || sexo == 'f';
    }

    private static boolean validarEstadoCivil(char estadoCivil) {
        char estado = Character.toUpperCase(estadoCivil);
        return estado == 'S' || estado == 'C' || estado == 'V' || estado == 'D';
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String nome;
        int idade;
        double salario;
        char sexo, estadoCivil;

        // Validação do nome
        do {
            System.out.print("Digite o nome (mais que 3 caracteres): ");
            nome = scanner.nextLine();
            if (!validarNome(nome)) {
                System.out.println("Nome inválido! Deve ter mais que 3 caracteres.");
            }
        } while (!validarNome(nome));

        // Validação da idade
        do {
            System.out.print("Digite a idade (entre 0 e 150): ");
            idade = scanner.nextInt();
            if (!validarIdade(idade)) {
                System.out.println("Idade inválida! Deve estar entre 0 e 150.");
            }
        } while (!validarIdade(idade));

        // Validação do salário
        do {
            System.out.print("Digite o salário (maior que zero): ");
            salario = scanner.nextDouble();
            if (!validarSalario(salario)) {
                System.out.println("Salário inválido! Deve ser maior que zero.");
            }
        } while (!validarSalario(salario));

        // Validação do sexo
        do {
            System.out.print("Digite o sexo (M/F): ");
            sexo = scanner.next().charAt(0);
            if (!validarSexo(sexo)) {
                System.out.println("Sexo inválido! Digite M ou F.");
            }
        } while (!validarSexo(sexo));

        // Validação do estado civil
        do {
            System.out.print("Digite o estado civil (S/C/V/D): ");
            estadoCivil = scanner.next().charAt(0);
            if (!validarEstadoCivil(estadoCivil)) {
                System.out.println("Estado civil inválido! Digite S, C, V ou D.");
            }
        } while (!validarEstadoCivil(estadoCivil));

        System.out.println("\nInformações validadas com sucesso!");
        System.out.println("Nome: " + nome);
        System.out.println("Idade: " + idade);
        System.out.println("Salário: " + salario);
        System.out.println("Sexo: " + sexo);
        System.out.println("Estado Civil: " + estadoCivil);

        scanner.close();
    }
}
